server/sonar-server/src/main/java/org/sonar/server/favorite/FavoriteUpdater.java

```diff
@@ -39,14 +43,30 @@ public FavoriteUpdater(DbClient dbClient, UserSession userSession) {
   /**
    * Set favorite to the logged in user. If no user is logged, no action is done
    */
-  public void add(DbSession dbSession, long componentDbId) {
+  public void add(DbSession dbSession, long componentDtoId) {
```

Comment:
IMO, parameters should be ComponentDto and UserDto, it will remove the needs of UserSession in this class.